src/app/pages/DetailsPage/detailsPage.tsx

```diff
@@ -1,34 +1,52 @@
-import { Link, useNavigate } from "react-router-dom";
+import { FormEvent } from "react";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import { Set } from "../../types";
 import ArrowIcon from "../../../assets/icons/ArrowIcon";
 import BinIcon from "../../../assets/icons/BinIcon";
 import HomeIcon from "../../../assets/icons/HomeIcon";
 import styles from "./detailsPage.module.css";
+import useCollection from "../../utils/useCollection";
+import deleteSet from "../../utils/deleteSet";
 
 export default function DetailsPage() {
   const navigate = useNavigate();
+  const params = useParams();
+
+  const sets = useCollection();
+  const collection: Set[] = sets.collection;
+  const set = collection.find((set) => set.numberSet == `${params.numberSet}`);
+
+  const deleteExistingSet = deleteSet(params);
+
+  const handleClick = async function (event: FormEvent) {
+    event.preventDefault();
+    await deleteExistingSet();
```

Comment:
I'd say that the `deleteSet` Function name already explains pretty well what it does, so it's not really necessary to make it another variable. 

```suggestion
  const handleClick = async function (event: FormEvent) {
    event.preventDefault();
    await deleteSet(params);
```

Reply:
unfortunately, when i put it that way, it doesn't work anymore